src/utils/cookies/cookieConsent.ts

```diff
@@ -0,0 +1,246 @@
+ * @see https://www.osano.com/cookieconsent/download/
+ */
+const init = (options: InitOptions): void => {
+  const {
+    allowedPages = null,
+    amplitudeInstance,
+    locale,
+    t,
+    theme,
+    userConsent,
+  } = options;
+  const { isUserOptedOutOfAnalytics, hasUserGivenAnyCookieConsent } = userConsent;
+  const { primaryColor } = theme;
+
+  if (!shouldDisplayConsentPopup(allowedPages)) {
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  require('cookieconsent'); // XXX Requiring it will make it available in the browser (cannot be used properly as a module)
+
+  // @ts-ignore
+  const cc = window.cookieconsent;
+  const additionalMessage = hasUserGivenAnyCookieConsent ? isUserOptedOutOfAnalytics ? t('cookieConsent.message.userDenied', `<br /><b>Vous aviez refusé les cookies. Veuillez reconfirmer votre choix.</b><br /><br />`) : t('cookieConsent.message.userAllowed', `<br /><b>Vous aviez autorisé les cookies. Veuillez reconfirmer votre choix.</b><br /><br />`) : '';
+  const message = t('cookieConsent.message.cookieUsage', `Nous utilisons des cookies pour faire des mesures d'audience anonymes afin de nous permettre d'identifier les fonctionnalités les plus utilisées et les pages les plus consultées.<br /><br /><i>Aucune information personnelle n'est recueillie, ni partagée à quelconque tiers.</i><br />${additionalMessage}`);
+
+  // Use https://www.osano.com/cookieconsent/download/ "Start Coding" to use the UI configuration builder
+  // See https://www.osano.com/cookieconsent/documentation/javascript-api/ for advanced API options and documentation
+  const cookieConsentSettings = {
+    // Behavior
+    autoOpen: true,
+    autoAttach: true,
+    type: 'opt-out', // We consider the user is opt-in by default and must opt-out manually to disable tracking
+    revokable: true, // Doesn't seem to work as expected, stuff gets revoked even when set to false, and also depends on the country
+    whitelistPage: [], // Doesn't seem to work at all, no visible effect
+    blacklistPage: [],
+    location: false, // XXX Can also be an object with advanced configuration to implement your own geolocation resolvers
+    cookie: {
+      name: CONSENT_COOKIE_NAME,
+      path: '/',
+      domain: window.location.hostname, // Uses the current domain
+      expiryDays: 365,
+      secure: process.env.NEXT_PUBLIC_APP_STAGE !== 'development', // Always use a secure cookie on non-dev stages
+    },
+    dismissOnScroll: false,
+    dismissOnTimeout: false, // XXX Beware there is a bug, buggy, will override previous choice stored in cookie
+    dismissOnWindowClick: false,
+
+    // UI (colors, visual design)
+    theme: 'classic',
+    position: 'bottom-right',
+    palette: {
+      popup: {
+        background: '#fff',
+      },
+      button: {
+        background: '#fff',
+        text: primaryColor,
+      },
+    },
+    // elements: {
```

Comment:
Why is this commented out? Are you planning on using it in the future? Maybe some comments here would help. :)

Reply:
Those are the default values, commented out for documentation purpose, I don't plan on using them, but it's much easier to try them out this way, because the official doc isn't so great.